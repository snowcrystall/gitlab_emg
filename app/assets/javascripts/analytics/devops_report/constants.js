import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Report');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Report gives an overview of how you are using GitLab from a feature perspective. Use it to view how you compare with other organizations.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
