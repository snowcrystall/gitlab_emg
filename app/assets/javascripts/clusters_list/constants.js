import { __, s__ } from '~/locale';

export const CLUSTER_ERRORS = {
  default: {
    tableText: s__('ClusterIntegration|Unknown Error'),
    title: s__('ClusterIntegration|Unknown Error'),
    description: s__(
      'ClusterIntegration|An unknown error occurred while attempting to connect to Kubernetes.',
    ),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your cluster status'),
      s__('ClusterIntegration|Make sure your API endpoint is correct'),
      s__(
        'ClusterIntegration|Node calculations use the Kubernetes Metrics API. Make sure your cluster has metrics installed',
      ),
    ],
  },
  authentication_error: {
    tableText: s__('ClusterIntegration|Unable to Authenticate'),
    title: s__('ClusterIntegration|Authentication Error'),
    description: s__('ClusterIntegration|GitLab failed to authenticate.'),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your token'),
      s__('ClusterIntegration|Check your CA certificate'),
    ],
  },
  connection_error: {
    tableText: s__('ClusterIntegration|Unable to Connect'),
    title: s__('ClusterIntegration|Connection Error'),
    description: s__('ClusterIntegration|GitLab failed to connect to the cluster.'),
    troubleshootingTips: [
      s__('ClusterIntegration|Check your cluster status'),
      s__('ClusterIntegration|Make sure your API endpoint is correct'),
    ],
  },
  http_error: {
    tableText: s__('ClusterIntegration|Unable to Connect'),
    title: s__('ClusterIntegration|HTTP Error'),
    description: s__('ClusterIntegration|There was an HTTP error when connecting to your cluster.'),
    troubleshootingTips: [s__('ClusterIntegration|Check your cluster status')],
  },
};

export const CLUSTER_TYPES = {
  project_type: __('Project'),
  group_type: __('Group'),
  instance_type: __('Instance'),
};

export const MAX_REQUESTS = 3;

export const STATUSES = {
  default: { className: 'bg-white', title: __('Unknown') },
  disabled: { className: 'disabled', title: __('Disabled') },
  created: { className: 'bg-success', title: __('Connected') },
  unreachable: { className: 'bg-danger', title: __('Unreachable') },
  authentication_failure: { className: 'bg-warning', title: __('Authentication Failure') },
  deleting: { title: __('Deleting') },
  creating: { title: __('Creating') },
};
