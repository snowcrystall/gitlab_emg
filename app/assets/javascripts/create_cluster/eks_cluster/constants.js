export const DEFAULT_REGION = 'us-east-2';

export const KUBERNETES_VERSIONS = [
  { name: '1.15', value: '1.15' },
  { name: '1.16', value: '1.16' },
  { name: '1.17', value: '1.17' },
  { name: '1.18', value: '1.18' },
  { name: '1.19', value: '1.19', default: true },
];
