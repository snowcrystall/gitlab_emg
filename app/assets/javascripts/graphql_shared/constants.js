export const TYPE_CI_RUNNER = 'Ci::Runner';
export const TYPE_EPIC = 'Epic';
export const TYPE_GROUP = 'Group';
export const TYPE_ISSUE = 'Issue';
export const TYPE_ITERATION = 'Iteration';
export const TYPE_ITERATIONS_CADENCE = 'Iterations::Cadence';
export const TYPE_MERGE_REQUEST = 'MergeRequest';
export const TYPE_MILESTONE = 'Milestone';
export const TYPE_PROJECT = 'Project';
export const TYPE_SCANNER_PROFILE = 'DastScannerProfile';
export const TYPE_SITE_PROFILE = 'DastSiteProfile';
export const TYPE_USER = 'User';
export const TYPE_VULNERABILITY = 'Vulnerability';
