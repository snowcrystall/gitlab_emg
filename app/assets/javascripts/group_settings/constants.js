import { __ } from '~/locale';

// Debounce delay in milliseconds
export const DEBOUNCE_TOGGLE_DELAY = 1000;

export const ERROR_MESSAGE = __('Refresh the page and try again.');

// runner setting options
export const ENABLED = 'enabled';
export const DISABLED = 'disabled_and_unoverridable';
export const ALLOW_OVERRIDE = 'disabled_with_override';
