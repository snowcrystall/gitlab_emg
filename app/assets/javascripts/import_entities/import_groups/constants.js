import { s__ } from '~/locale';

export const i18n = {
  NAME_ALREADY_EXISTS: s__('BulkImport|Name already exists.'),
};

export const NEW_NAME_FIELD = 'new_name';
