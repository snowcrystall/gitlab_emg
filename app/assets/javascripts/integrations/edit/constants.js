import { s__ } from '~/locale';

export const integrationLevels = {
  GROUP: 'group',
  INSTANCE: 'instance',
};

export const defaultIntegrationLevel = integrationLevels.INSTANCE;

export const overrideDropdownDescriptions = {
  [integrationLevels.GROUP]: s__(
    'Integrations|Default settings are inherited from the group level.',
  ),
  [integrationLevels.INSTANCE]: s__(
    'Integrations|Default settings are inherited from the instance level.',
  ),
};
