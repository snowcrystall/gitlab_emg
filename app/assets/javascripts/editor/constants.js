import { DEFAULT_DEBOUNCE_AND_THROTTLE_MS } from '~/lib/utils/constants';
import { __ } from '~/locale';

export const SOURCE_EDITOR_INSTANCE_ERROR_NO_EL = __(
  '"el" parameter is required for createInstance()',
);

export const URI_PREFIX = 'gitlab';
export const CONTENT_UPDATE_DEBOUNCE = DEFAULT_DEBOUNCE_AND_THROTTLE_MS;

export const ERROR_INSTANCE_REQUIRED_FOR_EXTENSION = __(
  'Source Editor instance is required to set up an extension.',
);

export const EDITOR_READY_EVENT = 'editor-ready';

export const EDITOR_TYPE_CODE = 'vs.editor.ICodeEditor';
export const EDITOR_TYPE_DIFF = 'vs.editor.IDiffEditor';

export const EDITOR_CODE_INSTANCE_FN = 'createInstance';
export const EDITOR_DIFF_INSTANCE_FN = 'createDiffInstance';

//
// EXTENSIONS' CONSTANTS
//

// For CI config schemas the filename must match
// '*.gitlab-ci.yml' regardless of project configuration.
// https://gitlab.com/gitlab-org/gitlab/-/issues/293641
export const EXTENSION_CI_SCHEMA_FILE_NAME_MATCH = '.gitlab-ci.yml';

export const EXTENSION_MARKDOWN_PREVIEW_PANEL_CLASS = 'md';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_PARENT_CLASS = 'source-editor-preview';
export const EXTENSION_MARKDOWN_PREVIEW_ACTION_ID = 'markdown-preview';
export const EXTENSION_MARKDOWN_PREVIEW_PANEL_WIDTH = 0.5; // 50% of the width
export const EXTENSION_MARKDOWN_PREVIEW_UPDATE_DELAY = 250; // ms
