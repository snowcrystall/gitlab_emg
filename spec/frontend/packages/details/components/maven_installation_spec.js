import { shallowMount, createLocalVue } from '@vue/test-utils';
import { nextTick } from 'vue';
import Vuex from 'vuex';
import { registryUrl as mavenPath } from 'jest/packages/details/mock_data';
import { mavenPackage as packageEntity } from 'jest/packages/mock_data';
import InstallationTitle from '~/packages/details/components/installation_title.vue';
import MavenInstallation from '~/packages/details/components/maven_installation.vue';
import { TrackingActions } from '~/packages/details/constants';
import CodeInstructions from '~/vue_shared/components/registry/code_instruction.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('MavenInstallation', () => {
  let wrapper;

  const xmlCodeBlock = 'foo/xml';
  const mavenCommandStr = 'foo/command';
  const mavenSetupXml = 'foo/setup';
  const gradleGroovyInstallCommandText = 'foo/gradle/groovy/install';
  const gradleGroovyAddSourceCommandText = 'foo/gradle/groovy/add/source';
  const gradleKotlinInstallCommandText = 'foo/gradle/kotlin/install';
  const gradleKotlinAddSourceCommandText = 'foo/gradle/kotlin/add/source';

  const store = new Vuex.Store({
    state: {
      packageEntity,
      mavenPath,
    },
    getters: {
      mavenInstallationXml: () => xmlCodeBlock,
      mavenInstallationCommand: () => mavenCommandStr,
      mavenSetupXml: () => mavenSetupXml,
      gradleGroovyInstalCommand: () => gradleGroovyInstallCommandText,
      gradleGroovyAddSourceCommand: () => gradleGroovyAddSourceCommandText,
      gradleKotlinInstalCommand: () => gradleKotlinInstallCommandText,
      gradleKotlinAddSourceCommand: () => gradleKotlinAddSourceCommandText,
    },
  });

  const findCodeInstructions = () => wrapper.findAll(CodeInstructions);
  const findInstallationTitle = () => wrapper.findComponent(InstallationTitle);

  function createComponent({ data = {} } = {}) {
    wrapper = shallowMount(MavenInstallation, {
      localVue,
      store,
      data() {
        return data;
      },
    });
  }

  afterEach(() => {
    wrapper.destroy();
  });

  describe('install command switch', () => {
    it('has the installation title component', () => {
      createComponent();

      expect(findInstallationTitle().exists()).toBe(true);
      expect(findInstallationTitle().props()).toMatchObject({
        packageType: 'maven',
        options: [
          { value: 'maven', label: 'Maven XML' },
          { value: 'groovy', label: 'Gradle Groovy DSL' },
          { value: 'kotlin', label: 'Gradle Kotlin DSL' },
        ],
      });
    });

    it('on change event updates the instructions to show', async () => {
      createComponent();

      expect(findCodeInstructions().at(0).props('instruction')).toBe(xmlCodeBlock);
      findInstallationTitle().vm.$emit('change', 'groovy');

      await nextTick();

      expect(findCodeInstructions().at(0).props('instruction')).toBe(
        gradleGroovyInstallCommandText,
      );
    });
  });

  describe('maven', () => {
    beforeEach(() => {
      createComponent();
    });

    it('renders all the messages', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('installation commands', () => {
      it('renders the correct xml block', () => {
        expect(findCodeInstructions().at(0).props()).toMatchObject({
          instruction: xmlCodeBlock,
          multiline: true,
          trackingAction: TrackingActions.COPY_MAVEN_XML,
        });
      });

      it('renders the correct maven command', () => {
        expect(findCodeInstructions().at(1).props()).toMatchObject({
          instruction: mavenCommandStr,
          multiline: false,
          trackingAction: TrackingActions.COPY_MAVEN_COMMAND,
        });
      });
    });

    describe('setup commands', () => {
      it('renders the correct xml block', () => {
        expect(findCodeInstructions().at(2).props()).toMatchObject({
          instruction: mavenSetupXml,
          multiline: true,
          trackingAction: TrackingActions.COPY_MAVEN_SETUP,
        });
      });
    });
  });

  describe('groovy', () => {
    beforeEach(() => {
      createComponent({ data: { instructionType: 'groovy' } });
    });

    it('renders all the messages', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('installation commands', () => {
      it('renders the gradle install command', () => {
        expect(findCodeInstructions().at(0).props()).toMatchObject({
          instruction: gradleGroovyInstallCommandText,
          multiline: false,
          trackingAction: TrackingActions.COPY_GRADLE_INSTALL_COMMAND,
        });
      });
    });

    describe('setup commands', () => {
      it('renders the correct gradle command', () => {
        expect(findCodeInstructions().at(1).props()).toMatchObject({
          instruction: gradleGroovyAddSourceCommandText,
          multiline: true,
          trackingAction: TrackingActions.COPY_GRADLE_ADD_TO_SOURCE_COMMAND,
        });
      });
    });
  });

  describe('kotlin', () => {
    beforeEach(() => {
      createComponent({ data: { instructionType: 'kotlin' } });
    });

    it('renders all the messages', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    describe('installation commands', () => {
      it('renders the gradle install command', () => {
        expect(findCodeInstructions().at(0).props()).toMatchObject({
          instruction: gradleKotlinInstallCommandText,
          multiline: false,
          trackingAction: TrackingActions.COPY_KOTLIN_INSTALL_COMMAND,
        });
      });
    });

    describe('setup commands', () => {
      it('renders the correct gradle command', () => {
        expect(findCodeInstructions().at(1).props()).toMatchObject({
          instruction: gradleKotlinAddSourceCommandText,
          multiline: true,
          trackingAction: TrackingActions.COPY_KOTLIN_ADD_TO_SOURCE_COMMAND,
        });
      });
    });
  });
});
