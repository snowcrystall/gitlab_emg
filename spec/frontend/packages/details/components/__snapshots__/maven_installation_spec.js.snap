// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MavenInstallation groovy renders all the messages 1`] = `
<div>
  <installation-title-stub
    options="[object Object],[object Object],[object Object]"
    packagetype="maven"
  />
   
  <code-instruction-stub
    class="gl-mb-5"
    copytext="Copy Gradle Groovy DSL install command"
    instruction="foo/gradle/groovy/install"
    label="Gradle Groovy DSL install command"
    trackingaction="copy_gradle_install_command"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy add Gradle Groovy DSL repository command"
    instruction="foo/gradle/groovy/add/source"
    label="Add Gradle Groovy DSL repository command"
    multiline="true"
    trackingaction="copy_gradle_add_to_source_command"
    trackinglabel="code_instruction"
  />
</div>
`;

exports[`MavenInstallation kotlin renders all the messages 1`] = `
<div>
  <installation-title-stub
    options="[object Object],[object Object],[object Object]"
    packagetype="maven"
  />
   
  <code-instruction-stub
    class="gl-mb-5"
    copytext="Copy Gradle Kotlin DSL install command"
    instruction="foo/gradle/kotlin/install"
    label="Gradle Kotlin DSL install command"
    trackingaction="copy_kotlin_install_command"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy add Gradle Kotlin DSL repository command"
    instruction="foo/gradle/kotlin/add/source"
    label="Add Gradle Kotlin DSL repository command"
    multiline="true"
    trackingaction="copy_kotlin_add_to_source_command"
    trackinglabel="code_instruction"
  />
</div>
`;

exports[`MavenInstallation maven renders all the messages 1`] = `
<div>
  <installation-title-stub
    options="[object Object],[object Object],[object Object]"
    packagetype="maven"
  />
   
  <p>
    <gl-sprintf-stub
      message="Copy and paste this inside your %{codeStart}pom.xml%{codeEnd} %{codeStart}dependencies%{codeEnd} block."
    />
  </p>
   
  <code-instruction-stub
    copytext="Copy Maven XML"
    instruction="foo/xml"
    label=""
    multiline="true"
    trackingaction="copy_maven_xml"
    trackinglabel="code_instruction"
  />
   
  <code-instruction-stub
    copytext="Copy Maven command"
    instruction="foo/command"
    label="Maven Command"
    trackingaction="copy_maven_command"
    trackinglabel="code_instruction"
  />
   
  <h3
    class="gl-font-lg"
  >
    Registry setup
  </h3>
   
  <p>
    <gl-sprintf-stub
      message="If you haven't already done so, you will need to add the below to your %{codeStart}pom.xml%{codeEnd} file."
    />
  </p>
   
  <code-instruction-stub
    copytext="Copy Maven registry XML"
    instruction="foo/setup"
    label=""
    multiline="true"
    trackingaction="copy_maven_setup_xml"
    trackinglabel="code_instruction"
  />
   
  <gl-sprintf-stub
    message="For more information on the Maven registry, %{linkStart}see the documentation%{linkEnd}."
  />
</div>
`;
